src/calc_FD_CG_2DES.c

```diff
@@ -0,0 +1,385 @@
+            read_doorway_window_from_file(non,"CG_2DES_windows_EA.dat",im_window_EA,re_window_EA,non->tmax1);
+            read_doorway_window_from_file(non,"CG_2DES_windows_GB.dat",im_window_GB,re_window_GB,non->tmax1);
+            read_doorway_window_from_file(non,"CG_2DES_windows_SE.dat",im_window_SE,re_window_SE,non->tmax1);
+            printf("Completed reading pre-calculated data.\n");
+        }
+        
+        call_final_FD_CG_2DES(non,P_DA,segments,re_doorway,im_doorway,
+                              re_window_SE,im_window_SE,
+                              re_window_GB,im_window_GB,
+                              re_window_EA,im_window_EA,
+                              re_2DES, im_2DES, Q1, Q2);
+        /*CG_full_2DES_segments(non,re_doorway,im_doorway,
+                              re_window_SE,im_window_SE,
+                              re_window_GB, im_window_GB,
+                              re_window_EA,im_window_EA,
+				                      P_DA,pro_dim);*/
+    }
+    
+    free(re_doorway),      free(im_doorway);
+    free(re_window_SE),    free(im_window_SE);
+    free(re_window_GB),    free(im_window_GB);
+    free(re_window_EA),    free(im_window_EA);  
+    free(P_DA);
+    free(Q1),free(Q2);
+    return;
+}
+
+/* =================================================== */
+/* COMBINE DOORWAY AND WINDOW FUNCTION FOR CG SEGMENTS */
+/* =================================================== */
+void FD_CG_full_2DES_segments(t_non *non,float *re_doorway,float *im_doorway,
+    float *re_window_SE,float *im_window_SE,float *re_window_GB, float *im_window_GB,
+    float *re_window_EA,float *im_window_EA,float *P_DA,int N, char *waittime,int wfile,
+    double *Q1, double *Q2){
+    int t1,t2,t3;
+    int S,R,W; // Segment indices
+    int indext1,indext2,indext3;
+    int sampleCount=1;
+    int pol,molPol;
+    int px[4];
+    int RR,RW;
+    float **re_2DES_NR_sum,  **re_2DES_R_sum, **im_2DES_NR_sum, **im_2DES_R_sum;
+    float polWeight;
+    float factorWEA1;
+    float factorWEA2;
+    float factorR;
+    float factorRR;
+
+    re_2DES_NR_sum = (float **)calloc2D(non->tmax3,non->tmax1,sizeof(float),sizeof(float*));
+    re_2DES_R_sum  = (float **)calloc2D(non->tmax3,non->tmax1,sizeof(float),sizeof(float*));
+    im_2DES_NR_sum = (float **)calloc2D(non->tmax3,non->tmax1,sizeof(float),sizeof(float*));
+    im_2DES_R_sum  = (float **)calloc2D(non->tmax3,non->tmax1,sizeof(float),sizeof(float*));
+    
+    /* We repeat everyting for each waiting time t2 */
+    //for (t2=0;t2<non->tmax2;t2++){
+    t2 = non->tmax2-1;
+    /* We repeat everyting for each labframe polarization */
+    for (pol=0;pol<3;pol++){
+      /* Loop over the 21 microscopic polarizations */
+	    for (molPol=0;molPol<21;molPol++){
+        polWeight=polarweight(pol,molPol);
+	      polar(px,molPol);	  
+	      /* Loop over times */
+	      for (t1=0;t1<non->tmax1;t1++){
+          for (t3=0;t3<non->tmax3;t3++){
```

Comment:
We do have a lot of for loops. It may be a lot of indentation, but better to do it.